octopusdeploy/schema_project_scheduled_trigger.go

```diff
@@ -0,0 +1,543 @@
+		} else if interval == filters.OnceEveryMinute {
+			minuteInterval := int16(continuousDailyScheduleFilterMap["minute_interval"].(int))
+			continuousDailyScheduleFilter.MinuteInterval = &minuteInterval
+		}
+
+		filter = continuousDailyScheduleFilter
+	}
+
+	if attr, ok := projectScheduledTrigger.GetOk("days_per_month_schedule"); ok {
+		if filter != nil {
+			return nil, errors.New("only one of 'once_daily_schedule', 'continuous_daily_schedule', 'days_per_month_schedule', or 'cron_expression_schedule' can be set")
+		}
+
+		daysPerMonthScheduleFilterList := attr.(*schema.Set).List()
+		daysPerMonthScheduleFilterMap := daysPerMonthScheduleFilterList[0].(map[string]interface{})
+
+		startTime, _ := time.Parse(filters.RFC3339NanoNoZone, daysPerMonthScheduleFilterMap["start_time"].(string))
+		monthlyScheduleType, _ := filters.MonthlyScheduleString(daysPerMonthScheduleFilterMap["monthly_schedule_type"].(string))
+
+		daysPerMonthScheduleFilter := filters.NewDaysPerMonthScheduledTriggerFilter(monthlyScheduleType, startTime)
+
+		daysPerMonthScheduleFilter.DateOfMonth = daysPerMonthScheduleFilterMap["date_of_month"].(string)
+		daysPerMonthScheduleFilter.DayNumberOfMonth = daysPerMonthScheduleFilterMap["day_number_of_month"].(string)
+
+		dayOfWeek, _ := filters.WeekdayString(daysPerMonthScheduleFilterMap["day_of_week"].(string))
+		daysPerMonthScheduleFilter.Day = &dayOfWeek
+
+		daysPerMonthScheduleFilter.TimeZone = timezone
+		filter = daysPerMonthScheduleFilter
+	}
+
+	if attr, ok := projectScheduledTrigger.GetOk("cron_expression_schedule"); ok {
+		if filter != nil {
+			return nil, errors.New("only one of 'once_daily_schedule', 'continuous_daily_schedule', 'days_per_month_schedule', or 'cron_expression_schedule' can be set")
+		}
+
+		cronExpressionScheduleFilterList := attr.(*schema.Set).List()
+		cronExpressionScheduleFilterMap := cronExpressionScheduleFilterList[0].(map[string]interface{})
+
+		cronExpression := cronExpressionScheduleFilterMap["cron_expression"].(string)
+		cronExpressionScheduleFilter := filters.NewCronScheduledTriggerFilter(cronExpression, timezone)
+
+		filter = cronExpressionScheduleFilter
+	}
+
+	// NewProjectTrigger doesn't actually use the description value
+	projectTriggerToCreate := triggers.NewProjectTrigger(name, description, isDisabled, project, action, filter)
+	projectTriggerToCreate.Description = description
+
+	return projectTriggerToCreate, nil
+}
+
+func getProjectScheduledTriggerSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": getNameSchema(true),
+		"description": {
+			Description: "A description of the trigger.",
+			Optional:    true,
+			Type:        schema.TypeString,
+		},
+		"project_id": {
+			Description:      "The ID of the project to attach the trigger.",
+			Required:         true,
+			Type:             schema.TypeString,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+			ForceNew:         true,
```

Comment:
What do these `ForceNew` attributes mean for the schema's properties?

Reply:
It causes the resource to be destroyed and recreated instead of updated when this property changes. So in this case if the project ID changes it will destroy it and recreate the trigger against the new project, without it will attempt to use the update endpoint which "succeeds" but the project doesn't actually change.

Reply:
Oh, is this a known issue with the update API then?